### DISCUSSION TITLE
Question on how to built a data processing workflow

### DISCUSSION THREAD

**Discussion by tomazk8** [elsa-workflows/elsa-core#405] [Q&A]
I am trying to build a workflow in code to process the data comming from sensors. I tried it using Fork and Join but could not find a way how to connect the activities to work. I have two data channels (producing one double value each) that I need to subtract, get an absolute, then compare this to a constant and output bool if value is above the limit. I want to use separate activities for these operations which are: Channel, Subtract (takes 2 inputs), Absolute, Constant and Compare (takes 2 inputs: value and a constant to compare to). Here is a flow for it:

```
Ch1 ->
Ch2 -> Subtract -> Absolute -> Compare -> [bool result]
                   Constant ->
```

Can you maybe help me with building a workflow in code, not the designer?

Thanks

**Comment by sfmskywalker** [comment]
That is a very nice use case.

The described flow can be implemented with a `Fork` and `Join` and continue with `Subtract` that accessed the output of `Ch1` and `Ch2`, although a more elegant solution would be to implement `Subtract` as a joinable activity (by subclassing the `Join` activity).

I'll create a new sample console project demonstrating the described flow. 

Besides that, I'm thinking that perhaps an even more elegant solution is if Elsa would natively support *any* activity to be configurable whether it should continue execution either *immediately* (as is the case today) or *after all* incoming activities have executed (i.e. native "join" behavior"). I'll put some more thought into this as a potential enhancement for Elsa 2.

**Comment by sfmskywalker** [comment]
I created [Sample25](https://github.com/elsa-workflows/elsa-core/tree/develop/src/samples/Sample25) that demonstrates the above flow.

As I was creating the example, I discovered a couple of subtle bugs. They have been fixed as well & published to NuGet (i.e. make sure to update any Elsa package references).

**Comment by tomazk8** [comment]
Thank you very much. I will try it out.
